internal/datacoord/import_checker.go

```diff
@@ -349,28 +354,36 @@ func (c *importChecker) checkStatsJob(job ImportJob) {
 	tasks := c.importMeta.GetTaskBy(c.ctx, WithType(ImportTaskType), WithJob(job.GetJobID()))
 	for _, task := range tasks {
 		originSegmentIDs := task.(*importTask).GetSegmentIDs()
-		statsSegmentIDs := task.(*importTask).GetStatsSegmentIDs()
+		sortSegmentIDs := task.(*importTask).GetStatsSegmentIDs()
 		taskCnt += len(originSegmentIDs)
 		for i, originSegmentID := range originSegmentIDs {
-			taskLogFields := WrapTaskLog(task, zap.Int64("origin", originSegmentID), zap.Int64("stats", statsSegmentIDs[i]))
-			t := c.si.GetStatsTask(originSegmentID, indexpb.StatsSubJob_Sort)
-			switch t.GetState() {
-			case indexpb.JobState_JobStateNone:
-				err := c.si.SubmitStatsTask(originSegmentID, statsSegmentIDs[i], indexpb.StatsSubJob_Sort, false)
+			taskLogFields := WrapTaskLog(task, zap.Int64("origin", originSegmentID), zap.Int64("target", sortSegmentIDs[i]))
+			originSegment := c.meta.GetHealthySegment(c.ctx, originSegmentID)
+			targetSegment := c.meta.GetHealthySegment(c.ctx, sortSegmentIDs[i])
+			if originSegment == nil {
+				// import zero num rows segment
+				doneCnt++
+				continue
+			}
+			if targetSegment == nil {
```

Comment:
a little bit hack, suggest to wrap a getSortCompactionTask function. And the process will be like:

has := hasSortCompactionTask
if !has {
  createSortCompactionTask
}